contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinWOOFi.sol

```diff
@@ -0,0 +1,139 @@
+
+    /// @dev Query the amount for selling the base token amount.
+    /// @param baseToken the base token to sell
+    /// @param baseAmount the amount to sell
+    /// @return quoteAmount the swapped quote amount
+    function querySellBase(
+        address baseToken, 
+        uint256 baseAmount
+    ) external view returns (uint256 quoteAmount);
+
+}
+
+contract MixinWOOFi{
+
+    using LibERC20TokenV06 for IERC20TokenV06;
+    using LibERC20TokenV06 for IEtherTokenV06;
+    using LibSafeMathV06 for uint256;
+
+    address constant rebateAddress = 0xBfdcBB4C05843163F491C24f9c0019c510786304;
+
+    // /// @dev Swaps an exact amount of input tokens for as many output tokens as possible.
+    // /// @param _amountIn Amount of input tokens to send
+    // /// @param _minAmountOut The minimum amount of output tokens that must be received for the transaction not to revert.
+    // /// @param _tokenIn Input token
+    // /// @param _tokenOut Output token
+    // /// @param _to recipient of tokens
+    // /// @param pool WOOFi pool where the swap will happen
+    function _swap(
+        uint _amountIn, 
+        address _tokenIn, 
+        address _tokenOut, 
+        IWooPP pool
+    ) internal returns(uint256 realToAmount){
+        uint256 realToAmount;
+        address quoteToken = pool.quoteToken();
+        if (_tokenIn == quoteToken) {
+            realToAmount = pool.sellQuote(
+                _tokenOut,
+                _amountIn,
+                1,
+                address(this),
+                rebateAddress
+            );
+        } else if (_tokenOut == quoteToken) {
+            realToAmount = pool.sellBase(
+                _tokenIn, 
+                _amountIn, 
+                1, 
+                address(this), 
+                rebateAddress
+            );
+        } else {
+            uint256 quoteTokenAmount = pool.querySellBase(_tokenIn, _amountIn);
+            IERC20TokenV06(pool.quoteToken()).approveIfBelow(address(pool), quoteTokenAmount);
+            
+            uint256 quoteAmount = pool.sellBase(
+                _tokenIn, 
+                _amountIn, 
+                0, 
+                address(this), 
+                rebateAddress
+            );
+            realToAmount = pool.sellQuote(
+                _tokenOut, 
+                quoteAmount, 
+                1, 
+                address(this), 
+                rebateAddress
+            );
+        }
+    }
+
+    function _tradeWOOFi(
+        IERC20TokenV06 sellToken,
+        IERC20TokenV06 buyToken,
+        uint256 sellAmount,
+        bytes memory bridgeData
+    )
+        public
+        returns (uint256 boughtAmount)
+    {
+        (IWooPP _pool) = abi.decode(bridgeData, (IWooPP));
+        uint256 beforeBalance = buyToken.balanceOf(address(this));
+
+        sellToken.approveIfBelow(address(_pool), sellAmount);
+        
+        //track the balance to know how much we bought
+        _swap(
```

Comment:
`_swap` returns the output amount. 

can we use the return value instead of doing the math based on before and after?

Reply:
When I do that, I just get a return value of 0.